apps/right-to-rent-check/behaviours/tenants.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = fields => {
+
+  return superclass => class extends superclass {
+
+    saveValues(req, res, callback) {
+      const name = req.body['tenant-name'];
+      const tenants = req.sessionModel.get('tenants') || [];
+
+      let tenant = _.find(tenants, {'tenant-name': name});
+      if (!tenant) {
+        tenant = fields.length ? _.pick(req.body, fields) : req.body;
+        if (!_.isEmpty(tenant)) {
```

Comment:
As discussed, I agree that we should generate a uid and use that instead of the name.

> That condition seems the wrong way around to me. "If the tenant already exists on the array, add it again". Does it need the ! removing?

The `!` is saying:
"If the tenant does NOT exist on the array, add it".